pkg/objstore/s3/s3.go

```diff
@@ -92,6 +94,11 @@ func parseConfig(conf []byte) (Config, error) {
 		config.PartSize = defaultMinPartSize
 	}
 
+	// here for testing
+	if config.Prefix != "" {
+		config.Prefix = strings.TrimSuffix(config.Prefix, DirDelim) + DirDelim
```

Comment:
Why is this needed?

Reply:
for the configuration parsing test

Reply:
Adding normal code just for tests doesn't sound like a good idea, consider removing this.

Reply:
Alright.